### ISSUE TITLE
How to modify/reconfigure Jackson object mapper used for response serialization

### ISSUE THREAD

**Issue by jshort** [dropwizard/dropwizard#2580]
I'm trying to add a Jackson FilterProvider to the environment's objectmapper (in @Override run())
```
environment.getObjectMapper()
                .setFilterProvider(new SimpleFilterProvider()
                                           .addFilter(MyFilter.NAME, MyFilter.INSTANCE));
```

And then I have a Jersey resource that returns type `Response`:

```
return Response.ok().entity(userCollection).build();
```

And I get a serialization error: `Cannot resolve PropertyFilter with id 'myFilter'; no FilterProvider configured`

Looking at the code, it seems that the `environment.getObjectMapper()` is passed to `protected Handler createAppServlet(` which calls `jersey.register(new JacksonBinder(objectMapper));` so it seems that this should work.

Relevant dependency versions:

DW: 1.3.4
Jackson: 2.9.5
Jersey: 2.25.1

**Comment by jshort** [comment]
Interesting: I injected my applications `Environment` into my jersey resource and then used that object mapper to serialize the object I pass to `Response.ok().entity(...)` and it serializes fine!  So something in the DW framework that takes the jax rs `Response` object and serializes it to clients is not using the `Environment.getObjectMapper()` it seems.

**Comment by jshort** [comment]
This is related to #1341 and @RobeEW's workaround fixed this problem.  Seems that some Jackson provider is getting found first?

**Comment by nickbabcock** [comment]
Ah, I wonder if it wouldn't be worth it to investigate a fix for #1341
